Mathlib/Algebra/Homology/HomotopyCategory/SingleFunctors.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.Algebra.Homology.HomotopyCategory.Shift
+import Mathlib.CategoryTheory.Shift.SingleFunctors
+
+/-!
+# Single functors from the homotopy category
+
+Let `C` be a preadditive category with a zero object.
+In this file, we put together all the single functors `C ⥤ CochainComplex C ℤ`
+along with their compatibilities with shifts into the definition
+`CochainComplex.singleFunctors C : SingleFunctors C (CochainComplex C ℤ) ℤ`.
+Similarly, we define
+`HomotopyCategory.singleFunctors C : SingleFunctors C (HomotopyCategory C (ComplexShape.up ℤ)) ℤ`.
+
+-/
+
+universe v' u' v u
+
+open CategoryTheory Category Limits
+
+variable (C : Type u) [Category.{v} C] [Preadditive C] [HasZeroObject C]
+
+namespace CochainComplex
+
+open HomologicalComplex
+
+/-- The collection of all single functors `C ⥤ CochainComplex C ℤ` along with
+their compatibilites with shifts. (This definition has purposely no `simps`
+attribute, as the generated lemmas would be very useful.) -/
```

Comment:
I don't think this is what you meant?
Or maybe even `irreducible_def` if you are 100% sure.

Reply:
I have fixed the phrasing. (`irreducible_def` would be too much.)